src/Reflection/EntityFieldsViaMagic.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace PHPStan\Reflection;
+
+use Drupal\Core\Entity\ContentEntityBase;
+use PHPStan\Type\MixedType;
+
+/**
+ * Allows field access via magic methods
+ *
+ * See \Drupal\Core\Entity\ContentEntityBase::__get and ::__set.
+ */
+class EntityFieldsViaMagic implements PropertiesClassReflectionExtension {
+
+  public function hasProperty(ClassReflection $classReflection, string $propertyName): bool {
+    return $classReflection->isSubclassOf(ContentEntityBase::class) &&
+      !$classReflection->hasNativeProperty($propertyName);
+  }
+
+  public function getProperty(ClassReflection $classReflection, string $propertyName): PropertyReflection {
+    return new class implements PropertyReflection {
+
+      public function getType(): \PHPStan\Type\Type {
+        return new MixedType();
```

Comment:
It's not a mixed type, we should return `FieldItemList`. I think it is _safe enough_ to assume no custom field type is writing a FieldItemListInterface implementation which builds off of Core. Worth noting that, however.